modules/rostests/apitests/shell32/ItemIDList.cpp

```diff
@@ -217,9 +234,32 @@ START_TEST(PIDL)
 
     pidl = SHCloneSpecialIDList(NULL, CSIDL_PRINTERS, FALSE);
     if (pidl)
-        TEST_CLSID(ILFindLastID(pidl), 0x71, 14, CLSID_Printers);
+    {
+        // Accept both the old and new format from the special folder API (NT5 vs NT6)
+        LPITEMIDLIST pidlLeaf = ILFindLastID(pidl);
+        if (LOBYTE(GetVersion()) < 6)
```

Comment:
Is the following code usable?
```suggestion
        if (!IsWindowsVistaOrGreater())
```

Reply:
Our shell32 just calls `GetVersion` and depends on PEB compat shims to do its thing. `IsWindows*` calls `VerifyVersionInfo` (IIRC).